coalib/results/TextRange.py

```diff
@@ -89,7 +89,31 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        first = False
+        second = False
+        if None not in (self.start.line, other.end.line):
+            if self.start.line == other.end.line:
+                if self.start.column is None or other.end.column is None:
+                    first = True
+                else:
+                    first = self.start.column <= other.end.column
+            else:
+                first = self.start.line < other.end.line
+        else:
+            first = True
+
+        if None not in (self.end.line, other.start.line):
+            if self.end.line == other.start.line:
+                if self.end.column is None or other.start.column is None:
+                    second = True
+                else:
+                    second = self.end.column >= other.start.column
+            else:
+                second = self.end.line > other.start.line
+        else:
+            second = True
+
+        return first and second
```

Comment:
actually this looks right... don't know myself how to simplify, it correctly accounts for the cases with `None`.... wouldn't have done it differently^^